app/services/moderation.py

```diff
@@ -185,3 +185,8 @@ async def get_count_auto_restrict(
         moderator=bot_user, user=target, chat=chat,
         type_restriction__in=(TypeRestriction.karmic_ro.name, TypeRestriction.karmic_ban.name),
     ).count()
+
+
+async def cancel_warn_user(moderator_event_id: int):
```

Comment:
We should consider naming this function something like 'delete_moderator_event', as it can be used for other moderator events besides just warnings.

Reply:
I think it would be better to create a new function delete_moderator_event, which is called in cancel_user_warn, but in handler we still call cancel_user_warn, wouldn't it?